packages/gradle-plugin/src/main/kotlin/io/realm/kotlin/gradle/RealmCompilerSubplugin.kt

```diff
@@ -237,6 +242,20 @@ class RealmCompilerSubplugin : KotlinCompilerPluginSupportPlugin, AnalyticsError
             options
         }
     }
+
+    /**
+     * Wrapper that ignores error if `failOnAnalyticsError=true`.
+     */
+    private fun <R> muteErrors(block: () -> R): R? {
```

Comment:
Why this needs to return?

Reply:
It is used guard collecting of target information and make them `null` if an error happens [here](https://github.com/realm/realm-kotlin/pull/1664/files#diff-aa43ef78792f7359466e6dafb449349572d491ed2f6f630cdd02e41880286075R220)